addons/misra.py

```diff
@@ -3185,6 +3187,38 @@ def misra_17_3(self, cfg):
         for w in cfg.clang_warnings:
             if w['message'].endswith('[-Wimplicit-function-declaration]'):
                 self.reportError(cppcheckdata.Location(w), 17, 3)
+        for token in cfg.tokenlist:
+            if token.str not in ["while", "if"]:
+                continue
+            if token.next.str != "(":
+                continue
+            tok = token.next
+            while (tok != token.next.link):
+                if tok.isName and not tok.function and not tok.valueType and tok.next.str == "(" and \
```

Comment:
hmm.. personally I rather use `is None` than `not` what do you think is more pythonic?